src/image/detail/MaskDict.cc

```diff
@@ -53,35 +84,49 @@ class MaskDict::GlobalState {
 
     ~GlobalState() = default;
 
+    std::mutex _mutex;
+    HandleList _allMaskDicts;  // all the live MaskDicts
+    std::shared_ptr<MaskDict> _defaultMaskDict;  // default MaskDict to use
 };
 
-std::shared_ptr<MaskDict> MaskDict::makeMaskDict() { return GlobalState::get().defaultMaskDict; }
+std::shared_ptr<MaskDict> MaskDict::makeMaskDict() {
+    GlobalState::Lock lock;
+    return lock.defaultMaskDict;
+}
 
 std::shared_ptr<MaskDict> MaskDict::makeMaskDict(MaskPlaneDict const& mpd) {
-    std::shared_ptr<MaskDict> dict = GlobalState::get().defaultMaskDict;
+    GlobalState::Lock lock;
+    std::shared_ptr<MaskDict> dict = lock.defaultMaskDict;
     if (!mpd.empty()) {
         dict = std::shared_ptr<MaskDict>(new MaskDict(mpd));
-        GlobalState::get().allMaskDicts.insert(dict);
+        lock.allMaskDicts.insert(dict);
```

Comment:
This is kind of an unfortunate choice of variable name, since the lock does not store all mask dicts...

Reply:
Yeah, maybe I should call the class `LockedProxy` and use `proxy` for its instances?  Or maybe your suggested refactor to move `get()` to lock will help address this, too.